ast/term.go

```diff
@@ -2897,6 +2916,14 @@ func unmarshalExpr(expr *Expr, v map[string]interface{}) error {
 			return fmt.Errorf("ast: unable to unmarshal negated field with type: %T (expected true or false)", v["negated"])
 		}
 	}
+	if generatedRaw, ok := v["generated"]; ok {
+		if b, ok := generatedRaw.(bool); ok {
+			expr.Generated = b
+		} else {
+			return fmt.Errorf("ast: unable to unmarshal generated field with type: %T (expected true or false)", v["generated"])
+		}
+	}
+
```

Comment:
[question/curiosity]: What motivated this case? :smiley: I'm always happy to see good defensive programming land in core utility functions, but this case is not entirely obvious from its surroundings. Is it caused by generated rules from `ast/compiler` or somesuch?

Reply:
My understanding on what generated: true means is elementary. However, I think it's that during compilation, some terms are rewritten and replaced with generated terms, perhaps to 'de-sugar' or some other reason.

Why this is here is that in order to round trip marshal and unmarshal an expression, this unmarshal logic was missing. It's not really related to the main functionality of this PR.